### PR TITLE
Delete SAP Monitor will also unassign and delete CSI if necessary

### PR DESCRIPTION
**Pull request by PakDLiu**
Delete SAP Monitor will also unassign and delete CSI if necessary
Create SAP Monitor will pass in KeyVault ID as well

### CODE REVIEW

File: azext_hanaonazure/custom.py [hunk 1]
```diff
@@ -130,9 +131,8 @@ def create_sapmonitor(
 
         # Create MSI
         msi_client = _msi_client_factory(cmd.cli_ctx)
-        arm_resource_id = arm_resource_id_format.format(msi_client.config.subscription_id, resource_group_name, monitor_name)
-        sapmon_id = 'sapmon-csi-{}'.format(hash(arm_resource_id))
-        msi = msi_client.user_assigned_identities.create_or_update(resource_group_name, sapmon_id, region)
+        csi_name = get_csi_name(msi_client.config.subscription_id, resource_group_name, monitor_name)
+        msi = msi_client.user_assigned_identities.create_or_update(resource_group_name, csi_name, region)
```
Comment (tniek) [inline azext_hanaonazure/custom.py]: Wouldn't this one be `csi`?

File: azext_hanaonazure/custom.py [hunk 2]
```diff
@@ -151,21 +151,67 @@ def create_sapmonitor(
...
+def delete_sapmonitor(cmd, client, resource_group_name, monitor_name):
+    sapmonitor = client.get(resource_group_name, monitor_name)
+    if sapmonitor.key_vault_id is not None and sapmonitor.hana_db_credentials_msi_id is not None:
+        # KeyVault ID found, unassigning CSI and deleting it
+        from ._client_factory import (_msi_client_factory, _keyvault_client_factory)
+        from azure.cli.core.profiles import ResourceType
+
+        # Get CSI
+        match = re.search(msi_id_format, sapmonitor.hana_db_credentials_msi_id)
+        msi_subscription_id = match.group(1)
```
Comment (tniek) [inline azext_hanaonazure/custom.py]: Same as above; wouldn't this wone be the `csi_subscription_id`?

**Review by pabowers — APPROVED** [review]
LGTM

**Review by rukawata — APPROVED** [review]

**Review by tniek — APPROVED** [review]
Overall looks good, just a few minor comments

**Review by persiaAziz-zz — APPROVED** [review]

**Pull request merged**